consult-lsp.el

```diff
@@ -322,6 +336,97 @@ CURRENT-WORKSPACE? has the same meaning as in `lsp-diagnostics'."
+                         flattened))
+        (remhash "children" table))))
+  flattened)
+
+(defun consult-lsp--file-symbols-candidates ()
+  "Returns all candidates for a `consult-lsp-file-symbols' search."
+  (consult--forbid-minibuffer)
+  (let ((all-symbols (consult-lsp--flatten-document-symbols
+                      (lsp-request "textDocument/documentSymbol"
+                                   (lsp-make-document-symbol-params :text-document
+                                                                    (lsp--text-document-identifier)))
+                      nil))
+        (candidates))
+    (dolist (symbol all-symbols)
+      (let ((line (thread-first symbol
+                    (lsp:document-symbol-selection-range)
+                    (lsp:range-start)
+                    (lsp:position-line)
+                    (lsp-translate-line)))
+            (cbeg (thread-first symbol
+                    (lsp:document-symbol-selection-range)
+                    (lsp:range-start)
+                    (lsp:position-character)
+                    (lsp-translate-column)))
+            (cend (thread-first symbol
+                    (lsp:document-symbol-selection-range)
+                    (lsp:range-end)
+                    (lsp:position-character)
+                    (lsp-translate-column))))
+        (let ((beg (lsp--line-character-to-point line cbeg))
+              (end (lsp--line-character-to-point line cend))
+              (marker (make-marker)))
+          (set-marker marker beg)
+          ;; Pre-condition to respect narrowing
+          (unless (or (< beg (point-min))
+                      (> end (point-max)))
+            (push (consult--location-candidate
+                   (consult--buffer-substring beg end 'fontify)
+                   marker
+                   (1+ line)
+                   'consult--type (lsp:symbol-information-kind symbol)
+                   'consult--name (lsp:symbol-information-name symbol))
+                  candidates)))))
+    (unless candidates
+      (user-error "No symbols"))
+    candidates))
+
+(defun consult-lsp--file-symbols-annotate ()
+  "Annotation function for `consult-lsp-file-symbols'."
+  (when consult-lsp-use-marginalia
+    (require 'marginalia))
+  (let* ((width (length (number-to-string (line-number-at-pos
+                                           (point-max)
+                                           consult-line-numbers-widen))))
+         (fmt (propertize (format "%%%dd " width) 'face 'consult-line-number-prefix)))
+    (lambda (cand)
+      (let ((line (cdr (get-text-property 0 'consult-location cand))))
+        (list cand (format fmt line)
+              (concat
+               (propertize (format " (%s)"
+                                   (alist-get (get-text-property 0 'consult--type cand)
+                                              lsp-symbol-kinds)) 'face 'font-lock-type-face)
+               (when consult-lsp-use-marginalia
```

Comment:
I guess that line should have the condition `(and consult-lsp-use-marginalia (bound-and-true-p 'marginalia-mode))`